Wox.Plugin.Translate/GoogleTranslatePlugin.cs

```diff
@@ -293,45 +284,36 @@ public void Init(PluginInitContext context)
             _context = context;
         }
 
-        public static int topThreadId;
         public List<Result> Query(Query query)
         {
-            topThreadId = Thread.CurrentThread.ManagedThreadId;
             _context.API.StartLoadingBar();
             var list = new List<Result>();
-            var actionParameters = query.ActionParameters;
-            if (actionParameters.Count <= 2)
+
+
+            if (query.ThirdSearch == string.Empty || query.ThirdSearch == " ")
             {
                 for (var index = 0; index < Languages.GetLength(0); ++index)
                 {
                     var str = Languages[index, 0];
                     var countrycode = Languages[index, 1];
-                    list.Add(new Result
-                    {
-                        Title = str,
-                        SubTitle = countrycode,
-                        IcoPath = "Images\\pic.png",
-                        Action = e =>
+                        list.Add(new Result
                         {
-                            _context.API.ChangeQuery(query.RawQuery + countrycode + " ", false);
-                            return false;
-                        }
-                    });
+                            Title = str,
+                            SubTitle = countrycode,
+                            IcoPath = "Images\\pic.png",
+                            Action = e =>
+                            {
+                                _context.API.ChangeQuery(query.RawQuery + countrycode + " ");
+                                return false;
+                            }
+                        });
                 }
             }
-            else if (actionParameters.Count > 2)
-            {
-                var input = actionParameters[2];
-                var from = actionParameters[0];
-                var to = actionParameters[1];
-                input = String.Join(" ", actionParameters.Skip(2).ToArray());
-                Thread.Sleep(500);
```

Comment:
The plugin had a Sleep here so that we don't attack Google's service for every letter we type. 
I understand that you removed this because it makes the plugin translate instantly for every character you type, but because of this plugin not using an official API I don't think attacking Google's server for every character that is being translated a good idea. Of course, how often the web is queried will depend on how fast you type, but it still is better to have a check in place.